src/v/redpanda/application.cc

```diff
@@ -459,7 +459,7 @@ void application::wire_up_services() {
                     : nullptr;
     syschecks::systemd_message("Starting internal RPC {}", rpc_cfg).get();
     construct_service(_rpc, rpc_cfg).get();
-
+    _deferred.emplace_back([this] { controller->shutdown_input().get(); });
```

Comment:
if this is placed here for its ordering w.r.t. other services, then i think it needs a comment (and so does the first phase). all the others are single phase and assumed to be ordered according to their start up sequence.